### ISSUE TITLE
Scatter Plot

### ISSUE THREAD

**Issue by DarioS** [broadinstitute/single_cell_portal#38]
It would be convenient to biologists to be able to select two genes and draw a scatterplot with each gene's expression values on one axis to look at co-expression. Perhaps also display a correlation coefficient in one of the plot corners where there are no points.

**Comment by vickylihorst** [comment]
Hi @DarioS, thanks for reaching out. To help us better understand your goal here, could you elaborate on why you would like a scatterplot for gene expressions? Is this part of a differential expression analysis?

**Comment by DarioS** [comment]
No. Rather than for differential expression (change in means) analysis, it will be useful for a *differential correlation* (dysregulation of network) analysis. An illustrative example is

![image](https://user-images.githubusercontent.com/631218/48109640-6578ef00-e29c-11e8-9129-d007a8e4a9bb.png)

The correlation between genes X and Y of single cell gene abundance values (each dot corresponds to a single cell) changes from about 0 to about 1 for B cells when samples from responders are considered, compared to the non-responders.

**Comment by vickylihorst** [comment]
Thanks for giving the example. Now we know you are looking for a correlation analysis, to help me better understand your request, could you dive deeper into why such feature is needed? 

In addition, where would this feature fit in your scientific inquiry: initial exploratory work, confirming a hypothesis, etc?

**Comment by DarioS** [comment]
I think it would be desirable to have because biologists often ask about the relationships between genes belonging to known gene networks, and that's [recently been shown to be more predictive than single genes in isolation](https://www.nature.com/articles/s41591-018-0157-9). I see the plot as being useful for confirming a result. Biologists might find a relationship between a pair of genes in their own data set and then use Single Cell Portal to quickly find a data set for the same or similar disease and hopefully find support for the generalisability of their result. Also, it's a kind of plot which [cBioPortal](http://www.cbioportal.org/) already has.

![image](https://user-images.githubusercontent.com/631218/48168334-eb4f7580-e342-11e8-9f1b-2d5daefae1b6.png)

**Comment by jlchang** [comment]
Hi @DarioS 

I've added a feature request for correlation plots to our product backlog. Thank you for using SCP and sharing your feedback!

Please don't hesitate to reach out if you have additional suggestions for improvements.
